vars/piperPipelineStageInit.groovy

```diff
@@ -103,15 +161,23 @@ void call(Map parameters = [:]) {
             if (parameters.script.commonPipelineEnvironment.configuration.runStep?.get('Init')?.slackSendNotification) {
                 slackSendNotification script: script, message: "STARTED: Job <${env.BUILD_URL}|${URLDecoder.decode(env.JOB_NAME, java.nio.charset.StandardCharsets.UTF_8.name())} ${env.BUILD_DISPLAY_NAME}>", color: 'WARNING'
             }
-            artifactSetVersion script: script
+            if (config.inferBuildTool && env.ON_K8S) {
+                artifactPrepareVersion script: script, buildTool: buildTool, dockerImage: ""
+            } else if (config.inferBuildTool) {
+                artifactPrepareVersion script: script, buildTool: buildTool
```

Comment:
Or you could add the resource definition to this parameter in the versioning.yml. As an example see [here](https://github.com/SAP/jenkins-library/blob/master/resources/metadata/cloudFoundryDeploy.yaml#L118). But it doesn't really matter.

Reply:
I wasn't aware that this exists. Would this have side effects for the general purpose pipeline if they decide to switch to `artifactPrepareVersion` and the cpe.buildTool is **not** set?

Reply:
I think not, because a GPP project would have to define `buildTool` in the general section of the config, and it is picked up from there with priority.